File: archive/j/javascript/remove-all-whitespace.js
const die = () => {
  console.log("Usage: please provide a string");
  process.exit(1);
}

if (process.argv.length != 3) {
  die();
}

process.argv[2].length > 0
  ? console.log(process.argv[2].replace(/\s/g, ''))
  : die();

File: archive/j/java/RemoveAllWhitespace.java
public class RemoveAllWhitespace {
    public static void main(String[] args) {
        if (args.length == 0 || args[0].isEmpty()) {
            showUsage();
        }

        System.out.println(removeAllWhitespace(args[0]));
    }

    private static void showUsage() {
        System.out.println("Usage: please provide a string");
        System.exit(1);
    }

    private static boolean isWhitespace(char c) {
        return switch (c) {
            case ' ', '\t', '\n', '\r' -> true;
            default -> false;
        };
    }

    private static String removeAllWhitespace(String input) {
        var result = new StringBuilder(input.length());

        for (char c : input.toCharArray()) {
            if (!isWhitespace(c)) {
                result.append(c);
            }
        }

        return result.toString();
    }
}